localstack/services/cloudformation/engine/template_deployer.py

```diff
@@ -1340,7 +1333,7 @@ def construct_changes(
         append_to_changeset=False,
         filter_unchanged_resources=False,
     ):
-        from localstack.services.cloudformation.provider import StackChangeSet
+        from localstack.services.cloudformation.engine.entities import StackChangeSet
```

Comment:
nit: due to the untangling in this PR, looks like we could move these imports to the top now